File: 383-RansomNote/383-RansomNote.java
// Last updated: 09/07/2026, 10:08:15
class Solution {
    public boolean canConstruct(String ransomNote, String magazine) {
        HashMap<Character, Integer> hm = new HashMap<>();
        for(int i = 0;i<magazine.length();i++){
            char ch = magazine.charAt(i);
            hm.put(ch,hm.getOrDefault(ch,0) + 1) ;
        }
        for(int i = 0;i<ransomNote.length();i++){
            char r = ransomNote.charAt(i);
            if(hm.containsKey(r) && hm.get(r)>0){
                hm.put(r,hm.get(r) - 1);
            }
            else{
                return false;
            }
        }
        return true;
    }
}

File: 383-RansomNote/383-RansomNote.py
# Last updated: 09/07/2026, 10:08:22
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        hm= {}
        for ch in magazine:
            hm[ch] = hm.get(ch,0) + 1
        for j in ransomNote:
            if hm.get(j,0) == 0:
                return False
            hm[j] -= 1
        return True      
            

        